packages/components/app/styles/components/side-nav/header.scss

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// SIDE-NAV > HEADER
+//
+
+@use "../../mixins/focus-ring" as *;
+
+@mixin hds-side-nav-icon-button($add-visible-border: false) {
+  color: var(--token-side-nav-foreground-primary-strong);
+  background-color: transparent;
+  border: 1px solid transparent;
+  border-radius: var(--token-side-nav-body-list-item-border-radius);
+  cursor: pointer;
+
+  @if ($add-visible-border) {
+    border-color: var(--token-color-palette-neutral-500);
+  }
+
+  &:focus,
+  &.mock-focus {
+    @include hds-focus-ring-with-pseudo-element($top: -1px, $right: -1px, $bottom: -1px, $left: -1px);
+  }
+
+  &:hover,
+  &.mock-hover {
+    color: var(--token-side-nav-foreground-primary-strong); // to avoid overrides by specificity (eg. `a:hover`)
+    background: var(--token-side-nav-interactive-surface-hover);
+  }
+
+  &:active,
+  &.mock-active {
+    color: var(--token-side-nav-foreground-primary-strong); // to avoid overrides by specificity (eg. `a:hover`)
+    background: var(--token-side-nav-interactive-surface-active);
+
+    @if ($add-visible-border) {
+      border-color: var(--token-color-palette-neutral-400);
+    }
+  }
+}
+
+// HEADER (LAYOUT WRAPPER)
+
+.hds-side-nav-header {
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+}
+
+
+// LOGO
+
+.hds-side-nav-header__logo-container {
+  display: flex;
+  flex: none;
+  align-items: center;
+  justify-content: center;
+  width: var(--token-side-nav-header-home-link-logo-size);
+  height: var(--token-side-nav-header-home-link-logo-size);
+}
+
+// "home-link"
+
+.hds-side-nav__home-link {
+  @include hds-side-nav-icon-button();
+  display: block;
+  width: 100%;
+  height: 100%;
+  padding: calc(var(--token-side-nav-header-home-link-padding) - 1px); // by design -  we take in account the transparent border
+}
+
+
+// ACTIONS
+
+.hds-side-nav-header__actions-container {
+  display: flex;
+
+  > * + * { margin-left: var(--token-side-nav-header-actions-spacing); }
+}
+
+// dropdown overrides
+
+.hds-side-nav__dropdown { // ⬅︎ add this class name to the `Hds::Dropdown` component to apply the style overrides
+  .hds-dropdown-toggle-icon {
+    @include hds-side-nav-icon-button($add-visible-border: true);
+  }
+}
+
+// generic "icon-button"
+
+.hds-side-nav__icon-button {
+  @include hds-side-nav-icon-button($add-visible-border: true);
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: 36px; // same height as the dropdown "toggle"
+  height: 36px;
```

Comment:
Would it make sense to create css custom properties for the values or, if you don't want to expose them, perhaps sass variables?